log-noise(api): "service config not found for azure:compute" on every recommendations read

## Summary

The API Lambda log group `/aws/lambda/cudly-dev-426fc8af-api` consistently emits this error on the recommendations read path:

```
[ERROR] failed to apply account overrides; returning un-filtered recs:
  resolve account configs: get service config azure/compute:
    service config not found for azure:compute
```

It happens during `ListRecommendations` when the override resolver tries to read the `(azure, compute)` `ServiceConfig` row from `service_configs` and finds none. The handler's fallback ("returning un-filtered recs") is correct — the user still sees Azure compute recommendations, just without per-account overrides applied — so the user impact is silent and behavioural rather than visible. But the log spam is a noise-floor issue and may indicate a missing default-seed migration.

## Reproduction

1. Open `/recommendations` on a CUDly deployment that has an Azure account configured.
2. Tail the API Lambda logs.
3. Every page load that includes Azure recs triggers the error line.

## Hypotheses to investigate

1. **Missing default-seed migration** for the `(azure, compute)` row. The repo has a migration sequence around `000040_split_savingsplans` that seeds `(aws, savings-plans-*)` rows from a previous umbrella row; a similar default-seed for `(azure, compute)` may have been omitted when Azure compute support landed.

2. **The resolver's "not found" path should be silent** if a missing config is genuinely the expected default-coverage state. If `(azure, compute)` is supposed to behave as "no overrides, use globals", the resolver should treat a missing row as a no-op rather than an error.

## Out of scope

This is not a #252 / #257 / #258 regression — the error has been emitted on every read since Azure recommendations support landed. Filing as its own follow-up rather than expanding any of those PRs.

## Acceptance

- Pick (1) or (2) above based on the project's design intent for default service configs
- If (1): add a migration that inserts the missing `(azure, compute)` default row
- If (2): change the resolver to log at debug level (or not at all) when the row is genuinely a "no overrides registered" no-op
- Either fix removes the recurring `[ERROR]` line from production Lambda logs

## Labels

Suggested: `type/bug`, `priority/p3`, `severity/low`, `urgency/eventually`, `impact/internal`, `effort/s`, `triaged`

--- a/internal/config/recommendation_overrides.go
+++ b/internal/config/recommendation_overrides.go
@@ -2,6 +2,7 @@ package config
 
 import (
 	"context"
+	"errors"
 	"fmt"
 )
 
@@ -48,6 +49,10 @@ func (c *globalConfigCache) lookup(ctx context.Context, store AccountConfigReade
 	}
 	fetched, err := store.GetServiceConfig(ctx, provider, service)
 	if err != nil {
+		if errors.Is(err, ErrNotFound) {
+			c.missing[k] = true
+			return nil, nil
+		}
 		return nil, fmt.Errorf("get service config %s/%s: %w", provider, service, err)
 	}
 	if fetched == nil {

--- a/internal/config/recommendation_overrides_test.go
+++ b/internal/config/recommendation_overrides_test.go
@@ -3,6 +3,7 @@ package config
 import (
 	"context"
 	"errors"
+	"fmt"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
@@ -253,4 +254,21 @@ func TestResolveAccountConfigsForRecs_OverrideErrorPropagates(t *testing.T) {
 	got, err := ResolveAccountConfigsForRecs(context.Background(), reader, recs)
 	assert.Error(t, err)
 	assert.Empty(t, got)
+}
+
+// TestResolveAccountConfigsForRecs_ErrNotFoundAbsorbed verifies that when
+// GetServiceConfig returns a wrapped ErrNotFound (the no-rows case), the
+// lookup treats it as absent rather than propagating it as an error. This
+// is the root scenario of issue #263: the INFO log firing on every refresh
+// for services that legitimately have no global config row.
+func TestResolveAccountConfigsForRecs_ErrNotFoundAbsorbed(t *testing.T) {
+	reader := &fakeAccountConfigReader{
+		globalErr: fmt.Errorf("service config not found for aws:rds: %w", ErrNotFound),
+	}
+	recs := []RecommendationRecord{acctRec("acct-A", "aws", "rds")}
+
+	got, err := ResolveAccountConfigsForRecs(context.Background(), reader, recs)
+	assert.NoError(t, err, "ErrNotFound from GetServiceConfig must not propagate as an error")
+	assert.Empty(t, got, "triple skipped when global is absent and no override")
+	assert.Equal(t, 1, reader.globalCalls, "global fetched once")
 }

--- a/internal/config/store_postgres.go
+++ b/internal/config/store_postgres.go
@@ -352,7 +352,7 @@ func (s *PostgresStore) GetServiceConfig(ctx context.Context, provider, service 
 
 	if err != nil {
 		if errors.Is(err, pgx.ErrNoRows) {
-			return nil, fmt.Errorf("service config not found for %s:%s", provider, service)
+			return nil, fmt.Errorf("service config not found for %s:%s: %w", provider, service, ErrNotFound)
 		}
 		return nil, fmt.Errorf("failed to get service config: %w", err)
 	}

--- a/internal/config/store_postgres.go
+++ b/internal/config/store_postgres.go
@@ -1833,7 +1833,7 @@ func (s *PostgresStore) GetActivePurchaseHistory(ctx context.Context, asOf time.
 // number, unknown provider) matches account_id with no provider gate. Providers
 // are sorted for deterministic SQL. The OR is wrapped in parentheses so it
 // composes with the surrounding AND chain.
-func appendAccountPredicate(conds []string, args []any, accountIDs []string, externalIDsByProvider map[string][]string) ([]string, []any) { //nolint:gocritic // unnamedResult: return names would conflict with body locals
+func appendAccountPredicate(conds []string, args []any, accountIDs []string, externalIDsByProvider map[string][]string) (outConds []string, outArgs []any) {
 	if len(accountIDs) == 0 && len(externalIDsByProvider) == 0 {
 		return conds, args
 	}

--- a/internal/config/store_postgres_mock_test.go
+++ b/internal/config/store_postgres_mock_test.go
@@ -131,7 +131,7 @@ func (s *testablePostgresStore) GetServiceConfig(ctx context.Context, provider, 
 
 	if err != nil {
 		if errors.Is(err, pgx.ErrNoRows) {
-			return nil, errors.New("service config not found")
+			return nil, fmt.Errorf("service config not found for %s:%s: %w", provider, service, ErrNotFound)
 		}
 		return nil, err
 	}

